test/integration/component/maint/testpath_disable_enable_host.py

```diff
@@ -0,0 +1,309 @@
+                snapshot on the same host
+            3. Try to reconnect the host :
+                -Host should get reconnected successfully
+        """
+        # Step 1
+        vm_root = VirtualMachine.create(
+            self.apiclient,
+            self.testdata["small"],
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id,
+        )
+        hostid = vm_root.hostid
+
+        vm_user = VirtualMachine.create(
+            self.userapiclient,
+            self.testdata["small"],
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id,
+        )
+
+        cmd = updateHost.updateHostCmd()
+        cmd.id = hostid
+        cmd.resourcestate = "Disabled"
+        cmd.allocationstate = "Disable"
+        self.apiclient.updateHost(cmd)
+        hostList = Host.list(self.apiclient, id=hostid)
+
+        self.assertEqual(hostList[0].resourcestate,
+                         "Disabled",
+                         "Check if the host is in disabled state"
+                         )
+        # Verify the exsisting vms should be running
+        self.assertEqual(vm_user.state,
+                         "Running",
+                         "Verify that the user vm is running")
+
+        self.assertEqual(vm_root.state,
+                         "Running",
+                         "Verify that the root vm is running")
+
+        vm_root.stop(self.apiclient)
+        vm_user.stop(self.apiclient)
+        root_state = self.dbclient.execute(
+            "select state from vm_instance where name='%s'" %
+            vm_root.name)[0][0]
+        user_state = self.dbclient.execute(
+            "select state from vm_instance where name='%s'" %
+            vm_user.name)[0][0]
+
+        self.assertEqual(root_state,
+                         "Stopped",
+                         "verify that vm should stop")
+
+        self.assertEqual(user_state,
+                         "Stopped",
+                         "verify that vm should stop")
+
+        VirtualMachine.create(
+            self.apiclient,
+            self.testdata["small"],
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id,
+            hostid=hostid)
+
+        root_volume = list_volumes(
+            self.apiclient,
+            virtualmachineid=vm_root.id,
+            type='ROOT',
+            listall=True
+        )
+
+        snap = Snapshot.create(
+            self.apiclient,
+            root_volume[0].id)
+
+        self.assertNotEqual(snap,
```

Comment:
No need of this assert. When snapshot creation fails, base.py won't return None, it will throw exception directly